go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/cluster-api-provider-aws/v2
 
-go 1.23.1
+go 1.24.7
```

Comment:
With this we need to stay to `go 1.24.0`, so always to the 0 patch release to avoid importing issues from other components